pydantic/_internal/_decorators.py

```diff
@@ -322,16 +324,27 @@ def gather_decorator_functions(cls: type[Any]) -> DecoratorInfos:
             shimmed_func = var_value.shim(func) if var_value.shim is not None else func
             info = var_value.decorator_info
             if isinstance(info, ValidatorDecoratorInfo):
-                res.validator[var_name] = Decorator(var_name, shimmed_func, func, info)
+                res.validator[var_name] = Decorator(cls.__name__, var_name, shimmed_func, func, info)
             elif isinstance(info, FieldValidatorDecoratorInfo):
-                res.field_validator[var_name] = Decorator(var_name, shimmed_func, func, info)
+                res.field_validator[var_name] = Decorator(cls.__name__, var_name, shimmed_func, func, info)
             elif isinstance(info, RootValidatorDecoratorInfo):
-                res.root_validator[var_name] = Decorator(var_name, shimmed_func, func, info)
+                res.root_validator[var_name] = Decorator(cls.__name__, var_name, shimmed_func, func, info)
             elif isinstance(info, FieldSerializerDecoratorInfo):
-                res.field_serializer[var_name] = Decorator(var_name, shimmed_func, func, info)
+                # check whether a serializer function is already registered for fileds
+                for field_ser_dec in res.field_serializer.values():
+                    # check serializers only in same model
```

Comment:
I would expand this comment to summarize the behavior as demonstrated by the tests below

Reply:
I've added more detail to the comment. let me know if it doesn't look good to you or suggest something